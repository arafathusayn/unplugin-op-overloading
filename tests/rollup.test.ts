import path from 'node:path'
import { rollupBuild, testFixtures } from '@sxzz/test-utils'
import { describe } from 'vitest'
import OperatorOverloading from '../src/rollup'

describe('rollup', async () => {
  const { dirname } = import.meta
  await testFixtures(
    '*.js',
    async (args, id) => {
      const { snapshot } = await rollupBuild(id, [OperatorOverloading()])
      return snapshot
    },
    { cwd: path.resolve(dirname, 'fixtures'), promise: true },
  )
})
