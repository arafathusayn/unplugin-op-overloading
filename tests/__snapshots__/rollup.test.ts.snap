// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rollup > basic.js 1`] = `
"// basic.js
const add = (() => {
  "operator-overloading disabled";
  const __lhs = a;
  const __rhs = b;
  const __sym = Symbol.for("+");
  return __lhs != null && __lhs[__sym] !== undefined
    ? __lhs[__sym](__rhs)
    : (__lhs + __rhs);
})();
const multiply = (() => {
  "operator-overloading disabled";
  const __lhs = x;
  const __rhs = y;
  const __sym = Symbol.for("*");
  return __lhs != null && __lhs[__sym] !== undefined
    ? __lhs[__sym](__rhs)
    : (__lhs * __rhs);
})();

export { add, multiply };
"
`;
